os/fs/driver/mtd/smart.c

```diff
@@ -4070,12 +4070,19 @@ static int smart_validate_crc(FAR struct smart_struct_s *dev)
 	if (crc != *((uint32_t *)header->crc32)) {
 		return -EIO;
 	}
-#else
+#elif defined(CONFIG_SMART_CRC_8)
 	/* Test 8-bit CRC for CRC8 & basic case for smart_scan. */
 	if (crc != header->crc8 && crc != SMART_ERASEDSTATE_CRC) {
 		return -EIO;
 	}
 
+#else
```

Comment:
smart_validate_crc is called when ioctl called with BIOC_READSECT. (read() of filesystem)
So, if fs try to read written value, this will be failed forever.

Reply:
- Added certain checks to make sure the basic CRC (in this case, only with logicalsector and sequence number) is not invalidated when the sector is in an erased state. (Do no try to match the crc values if header is in an erased state)

- If the calculated crc doesn't match the crc byte in the header, when the header is not in an erased state, an error signal will be returned. (If, the header crc byte matches the calculated crc ​OR the header is in an erased state, the crc shall be validated)

Reply:
I replied your mail, please refer to mine.